src/TableGateway/AbstractTableGateway.php

```diff
@@ -439,7 +439,10 @@ public function deleteWith(Delete $delete)
     protected function executeDelete(Delete $delete)
     {
         $deleteState = $delete->getRawState();
-        if ($deleteState['table'] != $this->table) {
+        if ($deleteState['table'] != $this->table
+            && (is_array($deleteState['table'])
```

Comment:
I think you want to have here `||` instead of `&&`. This should be covered in tests. I can't see related test to this change.

Reply:
In "executeSelect" method the same logic

Reply:
Hm... So I guess in both places it's wrong. What do you think?

Reply:
executeSelect for aliased table it works!

Reply:
@OleksandrBonar You are missing test to cover that change.

Reply:
@OleksandrBonar can you provide the unit test mentioned by @webimpress? Thanks!